inc/wplf-ajax.php

```diff
@@ -17,34 +18,58 @@ function wplf_ajax_submit_handler() {
   $return = apply_filters( 'wplf_validate_submission', $return );
 
   if( $return->ok ) {
-    // form existence has already been validated via filters
-
-    // copy $_POST to $form_data in order to undo WordPress' magic quotes
-    $form_data = stripslashes_deep($_POST);
+    // copy $_POST to $data in order to undo WordPress' magic quotes for
+    // wplf_post_data. Filtered data will be escaped appropriately.
+    $data = stripslashes_deep($_POST);
 
-    // Make form data filterable
-    $form_data = apply_filters( 'wplf_form_data', $form_data );
-
-    $form = get_post( intval( $form_data['_form_id'] ) );
+    // form existence has already been validated via filters
+    $form = get_post( intval( $data['_form_id'] ) );
+
+
+    // Make POST data filterable
+    $data = apply_filters( 'wplf_submission_data', $data );
+
+    $form_meta = get_post_meta( $form->ID );
+
+    // Compile all relevant data into a single array. This will be passed to wplf_post_validate_submission,
+    // so the same data wouldn't have to be looked up again.
+    // This is mostly for aesthetic reasons and DRY KISS.
+    $submission = [
+      'data' => $data,      // Filtered POST data
+      'meta' => [],         // Form metadata
+      'form' => $form,      // Form object
+      'post_title' => '',   // Filtered submission title
+    ];
+    // Seems that all metadata have only 1 value, so we simplify the structure a bit.
+    // TODO: Just to be sure, here's a reminder to check
+    foreach ($form_meta as $key => $value) {
+      $submission['meta'][$key] = $value[0];
+    }
 
-    $title_format = get_post_meta( $form->ID, '_wplf_title_format', true );
+    // Try _wplf_title_format from metadata but if it for some reason fails, just use form name
+    $title_format = !empty($submission['meta']['_wplf_title_format']) ? $submission['meta']['_wplf_title_format'] : $form->post_title;
```

Comment:
maybe break the ternary into separate lines? Can't see it completely without scrolling => tiny annoyance.